pkgs/development/tools/language-servers/crystalline/default.nix

```diff
@@ -47,6 +56,35 @@ crystal.buildCrystalPackage {
     wrapProgram "$out/bin/crystalline" --prefix PATH : '${lib.makeBinPath [ llvmPackages.llvm.dev ]}'
   '';
 
+  passthru = {
+    updateScript = _experimental-update-script-combinators.sequence [
+      (gitUpdater { rev-prefix = "v"; })
+      (_experimental-update-script-combinators.copyAttrOutputToFile "crystalline.shardLock" "${builtins.toString ./.}/shard.lock")
+      {
+        command = [
+          (writeShellScript "update-lock" "cd $1; ${lib.getExe crystal2nix}")
+          ./.
+        ];
+        supportedFeatures = [ "silent" ];
+      }
+      {
+        command = [
+          "rm"
+          "${builtins.toString ./.}/shard.lock"
+        ];
+        supportedFeatures = [ "silent" ];
+      }
+    ];
+    shardLock = runCommand "shard.lock" { inherit src; } ''
+      cp $src/shard.lock $out
+    '';
+
+    # Since doInstallCheck causes another test error, versionCheckHook is avoided.
```

Comment:
When enabled `doInstallCheck`, it caused following error

```
stdenv-linux/setup: line 1763
"unary operator expected"
```

It seems like a same error of https://github.com/NixOS/nixpkgs/pull/242681#discussion_r1261605348